airflow/www/views.py

```diff
@@ -760,22 +761,33 @@ def log(self, session=None):
             execution_date=execution_date, form=form,
             root=root, wrapped=conf.getboolean('webserver', 'default_wrap'))
 
-    @expose('/elasticsearch')
+    @expose('/redirect_to_external_log')
     @has_dag_access(can_dag_read=True)
     @has_access
     @action_logging
-    def elasticsearch(self):
+    @provide_session
+    def redirect_to_external_log(self, session=None):
         dag_id = request.args.get('dag_id')
         task_id = request.args.get('task_id')
         execution_date = request.args.get('execution_date')
+        dttm = timezone.parse(execution_date)
         try_number = request.args.get('try_number', 1)
-        elasticsearch_frontend = conf.get('elasticsearch', 'frontend')
-        log_id_template = conf.get('elasticsearch', 'log_id_template')
-        log_id = log_id_template.format(
-            dag_id=dag_id, task_id=task_id,
-            execution_date=execution_date, try_number=try_number)
-        url = 'https://' + elasticsearch_frontend.format(log_id=quote(log_id))
-        return redirect(url)
+
+        ti = session.query(models.TaskInstance).filter(
+            models.TaskInstance.dag_id == dag_id,
+            models.TaskInstance.task_id == task_id,
+            models.TaskInstance.execution_date == dttm).first()
+
+        try:
+            task_log_reader = TaskLogReader()
+            handler = task_log_reader.log_handler
+            url = handler.get_external_log_url(ti, try_number)
+            return redirect(url)
+        except AttributeError as e:
+            error_message = [
+                f"Task log handler does not support external log URLs.\n{str(e)}\n"
+            ]
+            return jsonify(message=error_message, error=True)
```

Comment:
I am not sure if this is the correct response format. This is not an API, but a view for the end-user, so the response should be text/html.

Reply:
I changed it to redirect to home with an error message, not sure if this is the right thing, let me know.